deployment/src/main/java/io/quarkiverse/qute/web/deployment/QuteWebTemplateBuildItem.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkiverse.qute.web.deployment;
+
+import static io.quarkiverse.qute.web.runtime.PathUtils.removeLeadingSlash;
+import static io.quarkiverse.qute.web.runtime.PathUtils.removeTrailingSlash;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+public final class QuteWebTemplateBuildItem extends MultiBuildItem {
+
+    /**
+     * templateId is used as path if link is null
+     */
+    private final String templateId;
+
+    /**
+     * The link to use for this template
+     */
+    private final String link;
+
+    public QuteWebTemplateBuildItem(String templateId, String link) {
```

Comment:
If I understand it correctly we make this build item part of the public API so we should add some info what the `templateId` and `link` should look like and how they're processed.

Also what happens if multiple `QuteWebTemplateBuildItem`s use the same link? Or if the link matches an existing path?

Reply:
THere is a duplicate key exception

Reply:
link has priority over path

Reply:
I am adding it